src/bp/nlu/engine/engine/predict-pipeline.ts

```diff
@@ -201,39 +199,42 @@ function MapStepToOutput(step: SpellStep): PredictOutput {
         value: s.slot.value,
         entity: entitiesMapper(s.slot.entity)
       }
-    }
+    ]
   }
 
-  const predictions: Predictions = step.ctx_predictions.intents.reduce((preds, current) => {
-    const { name: label, confidence } = current
+  const intentMapper = (intents: IntentPredictions): IntentPrediction[] => {
+    return intents.intents.map(intent => {
+      const slots: SlotPrediction[] =
+        step.slot_predictions_per_intent[intent.name]?.reduce(slotsCollectionReducer, []) || []
 
-    const intentPred = step.intent_predictions[label]
-    const intents = !intentPred
-      ? []
-      : intentPred.intents.map(({ name, confidence, extractor }) => ({
-          extractor,
-          label: name,
-          confidence,
-          slots: (step.slot_predictions_per_intent?.[name] || []).reduce(slotsCollectionReducer, {})
-        }))
+      return {
+        name: intent.name,
+        confidence: intent.confidence,
+        extractor: intent.extractor,
+        slots: _.orderBy(slots, s => s.confidence, 'desc')
+      }
+    })
+  }
 
-    return {
-      ...preds,
-      [label]: {
+  const contextMapper = (
+    ctxPredictions: IntentPredictions,
+    intentPredictions: _.Dictionary<NoneableIntentPredictions>
+  ): ContextPrediction[] => {
+    return ctxPredictions.intents.map(ctxPred => {
+      const { confidence, name } = ctxPred
+      const { intents, oos } = intentPredictions[name]
+      return {
+        name,
         confidence,
-        oos: intentPred?.oos || 0,
-        intents
+        intents: intentMapper({ intents }),
+        oos
       }
-    }
-  }, {})
+    })
+  }
 
   return {
-    entities,
-    predictions: _.chain(predictions) // orders all predictions by confidence
-      .entries()
-      .orderBy(x => x[1].confidence, 'desc')
-      .fromPairs()
-      .value(),
+    contexts: contextMapper(step.ctx_predictions, step.intent_predictions),
```

Comment:
This might be missing an `orderBy(ctx => ctx.confidence, 'desc')` so it keeps the exact same behavior